### PR TITLE
Add privacy considerations about UV, PII and User Handle

### LINKED ISSUE #578: Privacy Considerations should describe risks of storing userID/displayName in "second-factor" authenticators
After discussions about #558, there's agreement that in cases where the authenticator doesn't itself authenticate the user (e.g. via PIN, fingerprint, etc), it's a bad idea to store PII in the authenticator. The PII appears to be all of [MakePublicKeyCredentialOptions.user](https://w3c.github.io/webauthn/#dom-makepublickeycredentialoptions-user) except perhaps for `id` if the RP is careful (see below).

The RP is required to pass all of these fields to `create()` operations, so we should advise authenticators to avoid returning the non-`id` ones unless the authenticator has evidence that the same person is retrieving the fields as stored them.

We should also give guidance to RPs about what to store in user.id to make it minimally identifying. @leshi suggested that the RP should aggregate the ID with a random nonce and encrypt that with a key secret to the RP, so that people with access to the authenticator get what looks like random data.

### PR DESCRIPTION
**Pull request by emlun**
This fixes #578.

@equalsJeffH @selfissued Should we formally define "personally identifying information" (or, alternatively, reference an external definition)?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1041.html" title="Last updated on Aug 27, 2018, 10:51 AM GMT (1fe735d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1041/bd9bc3e...1fe735d.html" title="Last updated on Aug 27, 2018, 10:51 AM GMT (1fe735d)">Diff</a>

### CODE REVIEW

**Comment by emlun** [comment]
I tentatively marked this as technical at first, but I'll remove that label since #578 doesn't have it.

File: index.bs [hunk 1]
```diff
@@ -5401,6 +5407,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
 only to the operating system user that created that [=platform credential=].
 
 
+## Privacy of [PII] stored in authenticators ## {#sctn-pii-privacy}
+
+[=Authenticators=] MAY provide additional information to [=clients=] outside what's mandated by this specification &mdash; for
+example, to enable the [=client=] to provide a rich UI where the user can pick which [=credential=] to use for an [=authentication
```
Comment (selfissued) [inline index.bs]: "where" -> "with which"

File: index.bs [hunk 2]
```diff
@@ -5401,6 +5407,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
...
+ceremony=]. If an [=authenticator=] chooses to do so, it SHOULD NOT expose [PII] unless successful [=user verification=] has been
+performed. If the [=authenticator=] supports [=user verification=] with more than one concurrently enrolled user, the
+[=authenticator=] SHOULD NOT expose [PII] of other users than the currently [=user verified|verified=] user. Consequently, an
```
Comment (selfissued) [inline index.bs]: "other users than" -> "users other than"

File: index.bs [hunk 3]
```diff
@@ -5401,6 +5407,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
 only to the operating system user that created that [=platform credential=].
 
 
+## Privacy of [PII] stored in authenticators ## {#sctn-pii-privacy}
+
+[=Authenticators=] MAY provide additional information to [=clients=] outside what's mandated by this specification &mdash; for
```
Comment (selfissued) [inline index.bs]: " &mdash;" -> ","

File: index.bs [hunk 4]
```diff
@@ -5401,6 +5407,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
...
+For the purposes of this discussion, the [=user handle=] conveyed as the {{PublicKeyCredentialUserEntity/id}} member of
+{{PublicKeyCredentialUserEntity}} is not considered [PII].
+
+These recommendations serve to prevent an adversary with physical access to an [=authenticator=] from extracting [PII] about the
+owner(s) of the [=authenticator=].
+
+
+## User Handle contents ## {#sctn-user-handle-privacy}
+
+Since the [=user handle=] is not considered [PII] in [[#sctn-pii-privacy]], the [=[RP]=] SHOULD NOT include [PII] &mdash; for
```
Comment (selfissued) [inline index.bs]: " &mdash;" -> ","

File: index.bs [hunk 5]
```diff
@@ -5401,6 +5407,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
...
+example, e-mail addresses or usernames &mdash; in the [=user handle=]. This includes hash values of [PII], unless the hash
```
Comment (selfissued) [inline index.bs]: " &mdash;" -> ","

File: index.bs (hunk 3 above)
Comment (emlun) [reply index.bs]: Ok, I also changed "for example," to "e.g.," here to lessen the risk of the "for example" being read as a modifier for "Authenticators MAY provide additional information".

File: index.bs (hunk 4 above)
Comment (emlun) [reply index.bs]: Ditto

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs]: Thanks, done!

File: index.bs (hunk 2 above)
Comment (emlun) [reply index.bs]: Thanks, done!

File: index.bs (hunk 5 above)
Comment (emlun) [reply index.bs]: Thanks, done!

File: index.bs [hunk 6]
```diff
@@ -5416,6 +5422,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
 only to the operating system user that created that [=platform credential=].
 
 
+## Privacy of [PII] stored in authenticators ## {#sctn-pii-privacy}
+
+[=Authenticators=] MAY provide additional information to [=clients=] outside what's mandated by this specification, e.g.,
```
Comment (equalsJeffH) [inline index.bs]: nit: mandated => defined

File: index.bs [hunk 7]
```diff
@@ -5416,6 +5422,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
...
+to enable the [=client=] to provide a rich UI with which the user can pick which [=credential=] to use for an [=authentication
+ceremony=]. If an [=authenticator=] chooses to do so, it SHOULD NOT expose [PII] unless successful [=user verification=] has been
+performed. If the [=authenticator=] supports [=user verification=] with more than one concurrently enrolled user, the
+[=authenticator=] SHOULD NOT expose [PII] of users other than the currently [=user verified|verified=] user. Consequently, an
+[=authenticator=] that is not capable of [=user verification=] SHOULD NOT store [PII].
+
+For the purposes of this discussion, the [=user handle=] conveyed as the {{PublicKeyCredentialUserEntity/id}} member of
+{{PublicKeyCredentialUserEntity}} is not considered [PII].
```
Comment (equalsJeffH) [inline index.bs]: NIt:  [PII] => [PII], see [#sctn-user-handle-privacy].

File: index.bs [hunk 8]
```diff
@@ -5416,6 +5422,30 @@ authenticator=] and [=client device=] SHOULD work together to ensure that the ex
...
+
+These recommendations serve to prevent an adversary with physical access to an [=authenticator=] from extracting [PII] about the
+owner(s) of the [=authenticator=].
```
Comment (equalsJeffH) [inline index.bs]: owner(s) of the [=authenticator=] =>  [=authenticator=] 's enrolled user(s)

..?

File: index.bs (hunk 6 above)
Comment (emlun) [reply index.bs]: Done.

File: index.bs (hunk 7 above)
Comment (emlun) [reply index.bs]: Done.

File: index.bs (hunk 8 above)
Comment (emlun) [reply index.bs]: Looks good, thanks!

**Comment by equalsJeffH** [comment]
> Should we formally define "personally identifying information" (or, alternatively, reference an external definition)?

perhaps link to: https://en.wikipedia.org/wiki/Personally_identifiable_information ?  (the latter article addresses international PII definitions and legal regimes).

**Comment by nadalin** [comment]
@akshayku Please reviwe

**Comment by nadalin** [comment]
@emlun Please merge

**Pull request merged**